common/lib/container-definitions/src/error.ts

```diff
@@ -36,14 +38,30 @@ export enum ContainerErrorType {
 /**
  * Base interface for all errors and warnings at container level
  */
-export interface IErrorBase {
+export interface IErrorBase extends Error {
     /** errorType is a union of error types from
      * - container
      * - runtime
      * - drivers
      */
     readonly errorType: string;
+
+    /**
+     * See Error.message
+     * Privacy Note - This is a freeform string that we may not control in all cases (e.g. a dependency throws an error)
+     * If there are known cases where this contains privacy-sensitive data it will be tagged and included in the result
+     * of getTelemetryProperties. When logging, consider fetching it that way rather than straight from this field.
```

Comment:
See #6609 for additional thinking here.  I think this comment is sufficient, the concern is somewhat hypothetical.

Reply:
Can we generalize? I.e. say - all these fields (including stack) are only for local debugging. If you need to log anything to telemetry, you have to go through getTelemetryProperties() path and respect tags